components/clp-package-utils/clp_package_utils/scripts/native/search.py

```diff
@@ -132,26 +134,33 @@ def create_and_monitor_job_in_db(
 
             time.sleep(0.5)
 
-        with pymongo.MongoClient(results_cache.get_uri()) as client:
-            search_results_collection = client[results_cache.db_name][str(job_id)]
-            if max_num_results <= 0 or do_count_aggregation is not None:
-                cursor = search_results_collection.find()
-            else:
-                cursor = (
-                    search_results_collection.find().sort("timestamp", -1).limit(max_num_results)
-                )
-
-            if count_by_time_bucket_size is not None:
-                for document in cursor:
-                    print(f"timestamp: {document['timestamp']} count: {document['count']}")
-            elif do_count_aggregation is not None:
-                for document in cursor:
+        if do_count_aggregation is not None:
+            with pymongo.MongoClient(results_cache.get_uri()) as client:
+                search_results_collection = client[results_cache.db_name][str(job_id)]
+                for document in search_results_collection.find():
                     print(
                         f"tags: {document['group_tags']} count: {document['records'][0]['count']}"
                     )
-            else:
-                for document in cursor:
-                    print(f"{document['original_path']}: {document['message']}", end="")
+
+
+async def worker_connection_handler(reader: asyncio.StreamReader, writer: asyncio.StreamWriter):
+    try:
+        unpacker = msgpack.Unpacker()
+        while True:
+            # Read some data from the worker and feed it to msgpack
+            buf = await reader.read(1024)
+            if b"" == buf:
+                # Worker closed
+                return
+            unpacker.feed(buf)
+
+            # Print out any messages we can decode
+            for unpacked in unpacker:
+                print(f"{unpacked[0]}: {unpacked[2]}", end="")
+    except asyncio.CancelledError:
+        return
```

Comment:
What's the intent with swallowing the error?

Reply:
Do you think it's better to not catch the error and handle it in `main`?

Reply:
In this case, yeah, I think so since we're not doing anything special with the error and asyncio.run should swallow it anyway.